e2e/containers.e2e.spec.ts

```diff
@@ -0,0 +1,83 @@
+import path from 'path';
+import util from 'util';
+
+import { ElectronApplication, BrowserContext, _electron, Page } from 'playwright';
+import { test, expect } from '@playwright/test';
+import { createDefaultSettings, packageLogs, reportAsset, tool } from './utils/TestUtils';
+import { NavPage } from './pages/nav-page';
+
+let page: Page;
+
+/**
+ * Using test.describe.serial make the test execute step by step, as described on each `test()` order
+ * Playwright executes test in parallel by default and it will not work for our app backend loading process.
+ * */
+test.describe.serial('Containers Test', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item as the final option.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ],
+      env: {
+        ...process.env,
+        RD_LOGS_DIR: reportAsset(__filename, 'log'),
+      },
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: reportAsset(__filename) });
+    await packageLogs(__filename);
+    await electronApp.close();
+  });
+
+  test('wait for the container engine to be ready', async() => {
+    const navPage = new NavPage(page);
+
+    await navPage.progressBecomesReady();
+    await expect(navPage.progressBar).toBeHidden();
+  });
+
+  test('should run uname -m on containers from different architectures', async() => {
+    // Are we running moby or containerd? Run tests with that engine first,
+    // and then switch to the other.
+    const settingsOutput = await tool('rdctl', 'list-settings');
+    const settings = JSON.parse(settingsOutput);
+    const engine = settings.kubernetes.containerEngine;
+    const [otherEngine, commands] = engine === 'containerd' ? ['moby', ['nerdctl', 'docker']] : ['containerd', ['docker', 'nerdctl']];
+
+    // Run `uname -m` on two platforms using a more complex syntax
+    // nerdctl/docker run --rm --platform linux/amd64 --entrypoint uname busybox -m
+    expect(await tool(commands[0], 'run', '--rm', '--platform', 'linux/amd64', '--entrypoint', 'uname', 'busybox', '-m'))
+      .toContain('x86_64');
+    expect(await tool(commands[0], 'run', '--rm', '--platform', 'linux/arm64', '--entrypoint', 'uname', 'busybox', '-m'))
+      .toContain('aarch64');
+
+    await tool('rdctl', 'set', '--container-engine', otherEngine);
+    await util.promisify(setTimeout)(10_000);
```

Comment:
Please properly wait instead of having any fixed timeouts. (I mean, you _do_ properly wait, just also have a redundant timeout)